my_pyscf/dmet/_dmet.py

```diff
@@ -0,0 +1,577 @@
+        else:
+            dm_guess = get_basis_transform(dm, eo2ao.T)
+            veff = self.get_veff(eri, dm_guess)
+            nelecs = np.trace(dm_guess)
+            fock -= veff
+        
+        emb_mol = self._dummy_mol()
+        emb_mol.nelectron = round(nelecs)
+        emb_mol.max_memory = mf.mol.max_memory
+        emb_mol.build()
+
+        # Core energy contribution
+        core_energy = self._get_core_contribution(ao2eo=ao2eo, ao2co=ao2co)
+
+        if hasattr(mf, 'with_df') and mf.with_df is not None and self.density_fit:
+            emb_mf = scf.ROHF(emb_mol).density_fit()
+            emb_mf.with_df._cderi = eri
+        else:
+            emb_mf = scf.ROHF(emb_mol)
+            emb_mf._eri = eri
+
+        emb_mf.get_hcore = lambda *args: fock
+        emb_mf.get_ovlp  = lambda *args: np.eye(neo)
+        emb_mf.conv_tol = 1e-10
+        emb_mf.max_cycle = 100
+        emb_mf.energy_nuc = lambda *args: core_energy
+        emb_mf.kernel(dm_guess)
+
+        assert emb_mf.converged, 'DMET mean-field did not converge'
+
+        return emb_mf
+
+    def runDMET(self):
+        '''
+        Kernel to run the DMET calculations
+
+        1. Localize the orbitals
+        2. Fragment the molecule
+        3. Get the impurity subspace
+        4. Sanity checks for the number of electrons
+        5. Run the DMET mean-field
+        
+        Returns:
+            dmet_mf : SCF object
+                DMET mean-field object
+            ao2eo : np.array
+                Transformation matrix from AO to EO
+            ao2co : np.array
+                Core coefficient matrix
+        '''
+        self.do_localization_()
+        self.do_fragmentation_()
+        self.generate_impurity_subspace_()
+        self.get_imp_nelecs()
+        self.get_core_elecs()
+        self.dump_flags()
+        dmet_mf = self._get_dmet_mf()
+        return dmet_mf
+    
+    def _get_core_contribution(self, ao2eo=None, ao2co=None):
+        '''
+        Calculate the core energy
+        CoreEnergy = Nuclear Repulsion Energy + Energy from the core orbitals at the mean-field level.
+        '''
+        
+        mf = self.mf
+        neo = self.ao2eo.shape[-1]
+        s = mf.get_ovlp()
+        
+        if ao2eo is None:
+            ao2eo = self.ao2eo
+        if ao2co is None:
+            ao2co = self.ao2co
+        
+        '''
+        Core orbitals are doubly occupied therefore, can get 
+        away with this.
+        '''
+        dm_full_ao = dm = mf.make_rdm1()
+        if dm.ndim > 2:
+            dm = dm[0] + dm[1]
+       
+        cor2ao = ao2co.T @ s
+        core_dm = get_basis_transform(dm, cor2ao.T)
+       
+        core_dm = 0.5 * (core_dm + core_dm.T)
+
+        globalrdm = scipy.linalg.block_diag(np.zeros([neo, neo], dtype=core_dm.dtype), core_dm)
+       
+        # Convert the globalrdm to the AO basis
+        ao2eo = np.hstack([ao2eo, ao2co])
+        globalrdm1 = get_basis_transform(globalrdm, ao2eo.T)
```

Comment:
The essence of my earlier comment was that you are adding zero a whole lot here. Just do 
```
globalrdm = get_basis_transform (core_dm, ao2co.T)
```
It's the same thing, isn't it?